physlr/physlr.py

```diff
@@ -2236,24 +2237,216 @@ def physlr_bed_to_path(self):
+        Compute binomial estimation
+        """
+        mean_val = n * probability
+        std_dev = float(math.sqrt(mean_val * (1 - probability)))
+        return 0.5 * (1 + math.erf((x - mean_val) / (std_dev * math.sqrt(2))))
+
+    @staticmethod
+    def is_link_significant(ori_list):
+        """
+        Check if barcode link is strong enough
+        """
+        max_idx = ori_list.index(max(ori_list))
+        ori_list.sort()
+        max_val = ori_list[-1]
+        sum_with_second_max = ori_list[-1] + ori_list[-2]
+        normal_cdf = Physlr.normal_estimation(max_val, 0.5, sum_with_second_max)
+        if 1 - normal_cdf < 0.05:
+            return max_idx
+        return -1
+
+    @staticmethod
+    def orient_part_of_path_backward(pairs, path, unoriented, curr_pos, name):
+        """
+        Orient small part of path based on ARCS scaffold pairing information going backwards
+        """
+        idxtojoin = {0:"-+", 1:"--", 2:"++", 3:"+-"}
+        while unoriented:
+            prev_pos = curr_pos - 1
+            pair = (path[prev_pos][:-1], name[:-1])
+            if pair in pairs:
+                join_ori = pairs[pair]
+                max_idx = Physlr.is_link_significant(join_ori)
+                if max_idx != -1:
+                    curr_ori = idxtojoin[max_idx][1]
+                    if curr_ori == name[-1]:
+                        prev_ori = idxtojoin[max_idx][0]
+                        path[prev_pos] = path[prev_pos][0:-1] + prev_ori
+                        del unoriented[-1]
+                        curr_pos = prev_pos
+                        name = path[prev_pos]
+                    else:
+                        unoriented.clear()
+                else:
+                    unoriented.clear()
+            else:
+                unoriented.clear()
+
+        return path, unoriented
+
+
+    @staticmethod
+    def orient_part_of_path_forward(pairs, path, unoriented, curr_pos, name):
+        """
+        Orient small part of path based on ARCS scaffold pairing information going forwards
+        """
+        idxtojoin = {0:"-+", 1:"--", 2:"++", 3:"+-"}
+        prev_pos = curr_pos - 1
+        prev_name = path[prev_pos]
+        pair = (prev_name[:-1], name[:-1])
+        if pair in pairs:
+            join_orientation = pairs[pair]
+            max_idx = Physlr.is_link_significant(join_orientation)
+            if max_idx != -1:
+                if idxtojoin[max_idx][0] == prev_name[-1]:
+                    path[curr_pos] = name[:-1] + idxtojoin[max_idx][1]
+                else:
+                    unoriented.append(curr_pos)
+            else:
+                unoriented.append(curr_pos)
+        else:
+            unoriented.append(curr_pos)
+
+        return path, unoriented
+
+    @staticmethod
+    def orient_path(path, pairs):
+        """
+        Orient path based on ARCS scaffold pairing information
+        """
+        unoriented = []
+
+        if len(path) == 1 and path[0][-1] == ".":
```

Comment:
Yes.

Reply:
But shouldn't they be removed when the paths of sequences are generated?

Reply:
No, we don't remove paths of 1 length. In fact, when I have some time I need to look at that part of the code too. because I don't think we should be getting paths of length 1.

Reply:
Yes, I agree that these paths should be removed in the path-generation code. If you don't want to address that right now, it would be good if you made an issue so we can keep track of these future tasks

Reply:
I am against removing it even if it is length 1. Hypothetically speaking, when we have a scaffold the same size as a backbone, I want to know that backbone 0 mapped to scaffold 0. Furthermore, if we remove backbones, we no longer know which physlr scaffold is related to which physlr backbone. Right now line n = backbone n-1.

Reply:
Hmm ok I'm not sure I understand your most recent comment with your previous one then - I thought you said that we _should not_ be getting paths of length 1.

From an assembly improvement perspective, those paths don't have value. Which analyses on your end are dependent on that information? Why couldn't we still track which scaffold is based on which backbone without depending on that order? Is that information not in the header of the output scaffolds?